src/Nox.Types/Types/File/File.cs

```diff
@@ -1,9 +1,129 @@
+    }
+
+    /// <summary>
+    /// Creates an instance of the <see cref="File"/> class with specified options from the specified URL, pretty name and size in bytes.
+    /// </summary>
+    /// <param name="value">The value.</param>
+    /// <param name="options">The options.</param>
+    /// <returns></returns>
+    /// <exception cref="Nox.Types.TypeValidationException"></exception>
+    public static File From((string Url, string PrettyName, ulong SizeInBytes) value, FileTypeOptions options)
+    {
+        var newObject = new File
+        {
+            Value = value,
+            _fileTypeOptions = options
+        };
+
+        var validationResult = newObject.Validate();
+
+        if (!validationResult.IsValid)
+        {
+            throw new TypeValidationException(validationResult.Errors);
+        }
+
+        return newObject;
+    }
+
+    /// <summary>
+    /// Creates an instance of the <see cref="File"/> class with default options from the specified URL, pretty name and size in bytes.
+    /// </summary>
+    /// <param name="url">The URL of the file.</param>
+    /// <param name="prettyName">The pretty name of the file.</param>
+    /// <param name="sizeInBytes">The size of the file in bytes.</param>
+    /// <returns>An instance of the <see cref="File"/> class.</returns>
+    public static File From(string url, string prettyName, ulong sizeInBytes)
+        => From((url, prettyName, sizeInBytes));
+
+    /// <summary>
+    /// Creates an instance of the <see cref="File"/> class with specified options from the specified URL, pretty name and size in bytes.
+    /// </summary>
+    /// <param name="url">The URL of the file.</param>
+    /// <param name="prettyName">The pretty name of the file.</param>
+    /// <param name="sizeInBytes">The size of the file in bytes.</param>
+    /// <returns>An instance of the <see cref="File"/> class.</returns>
+    public static File From(string url, string prettyName, ulong sizeInBytes, FileTypeOptions options)
+        => From((url, prettyName, sizeInBytes), options);
+
+    /// <summary>
+    /// Retrieves the file format type extension from the URL.
+    /// </summary>
+    /// <returns>String representing the file format type extension.</returns>
+    public string GetFileExtension() => Path.GetExtension(new System.Uri(Url).LocalPath).ToLowerInvariant();
+
+    /// <summary>
+    /// Returns a string representation of the file value object.
+    /// </summary>
+    /// <returns>A string representation of the file value object.</returns>
+    public override string ToString()
+        => $"{PrettyName}: {Url}";
+
+    /// <summary>
+    /// Validates the <see cref="File"/> object.
+    /// </summary>
+    /// <returns>A validation result indicating whether the <see cref="File"/> object is valid or not.</returns>
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
+
+        System.Uri.TryCreate(Value.Url, UriKind.Absolute, out var x);
```

Comment:
What is this statement used for?

Reply:
That was left in by mistake. Removed it now, thanks